integration/client-cli/admin-cli/src/main/java/org/keycloak/client/admin/cli/KcAdmMain.java

```diff
@@ -79,24 +79,23 @@ public static void main(String [] args) {
             ArrayList<String> arguments = new ArrayList<>();
             arguments.add("kcadm");
             arguments.addAll(Arrays.asList(args));
+            int realmIndex = arguments.indexOf("--realm");
+            if (realmIndex != -1 && realmIndex < arguments.size() - 1) {
+                arguments.set(realmIndex + 1, arguments.get(realmIndex + 1).replace("<", "%3C").replace(">", "%3E"));
```

Comment:
Afaik, we are not doing the explicit conversion anywhere else, in server terms we leave it to Quarkus (specifically UriHandler in rest-easy-reactive-common) to handle (and transform) it, see:
![image](https://github.com/keycloak/keycloak/assets/12138171/fa12dbf8-0cff-4bde-a1ca-842e38c015af)

This it the [place](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/RealmsAdminResource.java#L147) where we get the escaped value which is then used in the well-known endpoints within the administration console.

Currently, if the raw angle brackets are left in, it causes the following error:
```
Logging into http://localhost:8080 as user <> of realm <>
Illegal character in path at index 29: http://localhost:8080/realms/<>/protocol/openid-connect/token
Try 'kcadm.sh help config credentials' for more information
```

That's the reason I transformed it.

Reply:
> Currently, if the raw angle brackets are left in, it causes the following error:

If I understand correctly the path method here:
```AdminRoot.realmsUrl(session.getContext().getUri()).path(realm.getName()).build()```
is generally handling reserved characters.  However <> since they aren't strictly forbidden, they are still allowed.  That is different than the behavior you are showing above with the UriHandler, so I'd say this should be a bug in resteasy, or whatever is backing the implenentation of the UriBuilder path method because <> should not be allowed in path segments, only the query portion.

If you url encode the <> won't the path method just enode that further?  % -> %3C -> %253C in which case it still won't be interpretted correctly.

Reply:
> If you url encode the <> won't the path method just enode that further?
No, here I created a realm named %3C and this is the result:
![image](https://github.com/keycloak/keycloak/assets/12138171/b8cf2542-9f3d-4ce6-b864-afc3f53b9317)

The difference in processing can be spotted in this method [1]. So that shouldn't be a problem.

But there definitely is a bug and I spotted it once I tried to create a realm named %3C. The console allows you to create it but when you try to access Realm settings, this happens (notice that the value is transformed to < in the url):
![image](https://github.com/keycloak/keycloak/assets/12138171/bb1d585b-c3bb-482b-b678-fe3ccd805a91)

I went further and created a realm named < right afterwards. Then I switched back to the previous %3C realm and tried to access the setting again. The error is gone but only because I got redirected from %3C to < realm immediately after I click on Realm settings.

I think that this certainly deserves a follow-up, since I think it already went beyond the scope of this PR.

cc: @vmuzikar

[1] https://github.com/quarkusio/quarkus/blob/main/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/util/Encode.java#L398